### PR TITLE
[CEDS-2886] Goods Location Code page changes

### PR DESCRIPTION
**Pull request by glidester**
Change title and content on page

Create a new expander that assists users in
finding the correct location codes.

### CODE REVIEW

File: app/views/components/goodsLocationExpander.scala.html [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+@import config.AppConfig
+
+@this(govukDetails : GovukDetails, link: link, paragraph: paragraphBody, appConfig: AppConfig)
+
+@()(implicit messages: Messages)
+
+@expanderContent = {
+    @paragraph(messages("declaration.goodsLocation.expander.intro"))
+
+    <p><b>@messages("declaration.goodsLocation.expander.paragraph1.title")</b></p>
```
Comment (udinobi) [inline app/views/components/goodsLocationExpander.scala.html:28]: I did not verify it but I think you could also write this like:
```
@paragraph(messages("declaration.goodsLocation.expander.paragraph1.title"), "govuk-body bold")
```
Maybe you can also omit the `govuk-body` class.

File: app/views/components/goodsLocationExpander.scala.html [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+    <div id="code-hint" class="govuk-hint">
+    @Html(messages("declaration.goodsLocation.expander.paragraph1.text",
+        link(
+            text = messages("declaration.goodsLocation.expander.paragraph1.link1.text"),
+            call = Call("GET", appConfig.companyInformationRegister),
+            target = "_new"
```
Comment (udinobi) [inline app/views/components/goodsLocationExpander.scala.html:34]: `_new` ? Do you have an `iframe` in this page? :thinking:

File: app/views/components/goodsLocationExpander.scala.html (hunk 1 above)
Comment (glidester) [reply app/views/components/goodsLocationExpander.scala.html]: Ok, yeah that works :+1:

File: app/views/components/goodsLocationExpander.scala.html (hunk 2 above)
Comment (glidester) [reply app/views/components/goodsLocationExpander.scala.html]: iframe?? Not sure about that but back in the day "_new" was considered ok, looks like things have changed though as I've now read it must be "_blank" according to HTML5 spec. Updated accordingly :slightly_smiling_face:

File: app/views/declaration/goods_location.scala.html [hunk 3]
```diff
@@ -59,10 +60,10 @@
             sectionHeaderKey = None
         )
 
+        @goodsLocationExpander()
 
         @tariffExpander(GoodsLocationForm, request.declarationType)(messages, appConfig)
```
Comment (MaciejRewera) [inline app/views/declaration/goods_location.scala.html:65]: Shouldn't `tariffExpander` be removed?

**Review by MaciejRewera — COMMENTED** [review]
I'd suggest adding specs for the new component. It is "rich in content" so I'd say it deserves a dedicated test suite.

File: app/views/declaration/goods_location.scala.html (hunk 3 above)
Comment (glidester) [reply app/views/declaration/goods_location.scala.html:65]: No! Liz was very specific about that :smile:

File: app/views/components/goodsLocationExpander.scala.html [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+    @paragraph(messages("declaration.goodsLocation.expander.paragraph1.title"), "bold")
+    <div id="code-hint" class="govuk-hint">
+    @Html(messages("declaration.goodsLocation.expander.paragraph1.text",
+        link(
+            text = messages("declaration.goodsLocation.expander.paragraph1.link1.text"),
+            call = Call("GET", appConfig.companyInformationRegister),
+            target = "_blank"
+        ).body
+    ))
+    </div>
```
Comment (udinobi) [inline app/views/components/goodsLocationExpander.scala.html:37]: Maybe the entire `<div...>...</div>` can be replaced with something like:
```
@govukHint(Hint(
    content = Html(messages("declaration.goodsLocation.expander.paragraph1.text",
        link(
            text = messages("declaration.goodsLocation.expander.paragraph1.link1.text"),
            call = Call("GET", appConfig.companyInformationRegister),
            target = "_blank"
        )
    ))
))
```

Comment (glidester) [reply app/views/components/goodsLocationExpander.scala.html]: Done :+1:

**Comment by glidester** [comment]
> I'd suggest adding specs for the new component. It is "rich in content" so I'd say it deserves a dedicated test suite.

IMO as there is no behavior to the code (merely presenting content with no dynamic elements) there is no benefit to writing a unit test for it.

**Review by udinobi — APPROVED** [review]

**Pull request merged**